examples/cfd/xaeronet/surface/train.py

```diff
@@ -0,0 +1,383 @@
+        validation_ids = [id[0] for _, id in validation_dataloader]
+        print(f"Training dataset size: {len(graphs)*dist.world_size}")
+        print(f"Validation dataset size: {len(validation_dataloader)}")
+
+    ######################################
+    # Training #
+    ######################################
+
+    # Initialize model
+    model = MeshGraphNet(
+        input_dim_nodes=6,
+        input_dim_edges=4,
+        output_dim=4,
+        processor_size=cfg.num_message_passing_layers,
+        aggregation="sum",
+        hidden_dim_node_encoder=256,
+        hidden_dim_edge_encoder=256,
+        hidden_dim_node_decoder=256,
+        mlp_activation_fn="relu",
+        do_concat_trick=True,
+        num_processor_checkpoint_segments=3,
+    ).to(device)
+    print(f"Number of trainable parameters: {count_trainable_params(model)}")
+
+    # DistributedDataParallel wrapper
+    if dist.world_size > 1:
+        model = DistributedDataParallel(
+            model,
+            device_ids=[dist.local_rank],
+            output_device=dist.device,
+            broadcast_buffers=dist.broadcast_buffers,
+            find_unused_parameters=dist.find_unused_parameters,
+            gradient_as_bucket_view=True,
+            static_graph=True,
+        )
+
+    # Optimizer and scheduler
+    optimizer = optim.Adam(model.parameters(), lr=0.001)
+    scheduler = optim.lr_scheduler.CosineAnnealingLR(
+        optimizer, T_max=2000, eta_min=1e-6
+    )
+    scaler = GradScaler()
+    print("Instantiated the model and optimizer")
+
+    # Check if there's a checkpoint to resume from
+    start_epoch, _ = load_checkpoint(
+        model, optimizer, scaler, scheduler, cfg.checkpoint_filename
+    )
+
+    # Training loop
+    print("Training started")
+    for epoch in range(start_epoch, cfg.num_epochs):
+        model.train()
+        total_loss = 0
+        for i in range(len(graphs)):
+            optimizer.zero_grad()
+            subgraphs = graphs[i]  # Get the partitions of the graph
+            for j in range(cfg.num_partitions):
+                with torch.autocast(amp_device, enabled=True, dtype=amp_dtype):
+                    part = subgraphs[j].to(device)
+                    ndata = torch.cat(
+                        (part.ndata["coordinates"], part.ndata["normals"]), dim=1
+                    )
+                    pred = model(ndata, part.edata["x"], part)
+                    pred_filtered = pred[part.ndata["inner_node"].bool(), :]
+                    target = torch.cat(
+                        (part.ndata["pressure"], part.ndata["shear_stress"]), dim=1
+                    )
+                    target_filtered = target[part.ndata["inner_node"].bool()]
+                    loss = (
+                        torch.mean((pred_filtered - target_filtered) ** 2)
+                        / cfg.num_partitions
+                    )
+                    total_loss += loss.item()
+                scaler.scale(loss).backward()
```

Comment:
I was wondering since this is similar to gradient accumulation, would it make sense to use DDP `no_sync` here to improve the perf?

Reply:
I tried this but unfortunately I can't get it to work properly with AMP. I'll explore this further and open another PR once I fix the issue with AMP.